src/index.ts

```diff
@@ -766,17 +768,40 @@ export default class Oas {
     return this.api?.[extension];
   }
 
+  /**
+   * Retrieve any circular `$ref` pointers that maybe present within the API definition.
+   *
+   * This method requires that you first dereference the definition.
+   *
+   * @see Oas.dereference
+   */
+  getCircularReferences() {
+    if (!this.dereferencing.complete) {
+      throw new Error('#dereference() must be called first in order for this method to obtain circular references.');
+    }
+
+    return this.dereferencing.circularRefs;
+  }
+
   /**
    * Dereference the current OAS definition so it can be parsed free of worries of `$ref` schemas
    * and circular structures.
    *
    */
   async dereference(
     opts: {
+      /**
+       * A callback method can be supplied to be called when dereferencing is complete. Used for
+       * debugging that the mulit-promise handling within this method works.
+       *
+       * @private
+       */
+      cb?: () => void;
```

Comment:
I don't like having this option here just for testing purposes but because I changed way we use `@readme/json-schema-ref-parser` to instantiate it with a class constructor it's really hard with Jest to spy on `new $RefParser.dereference()` now to ensure that it's only ever called once.

Adding this callback in here, and then calling it in the `.then()` below replicates this behavior.